python/tests/test_nearest_neighbors.py

```diff
@@ -364,10 +369,18 @@ def test_nearest_neighbors(
             item_df_withid.select(alias.row_number).toPandas()[alias.row_number]
         )
 
-        # test kneighbors: compare distances
+        # test kneighbors: compare squared distances
+        # note that single node and multi node may run slightly different kernels resulting
+        # in different distances.  This is especially an issue for self distances which don't come out
+        # to be 0 necessarily due to expanded form of calculation (|x-y|^2 = |x| + |y| - 2 <x,y>).
```

Comment:
Not quite understand but feel |x-y|^2 = `|x|^2` + `|y|^2` - 2 <x,y>. Or it should be `|x|`, `|y|` instead of `|x|^2` and `|y|^2`. 

Seems like `|x|` and `|y|` are used to approximate their squared value.